### DISCUSSION TITLE
How do we avoid losing macro-particles / resolution with the moving window?

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#639] [Q&A]
Hello, 

- My set up is 2 ion beam propagating through a background plasma with an exponentially decaying density profile. I think this density profile means that as the window moves from the left side more background electrons & ions are removed than they are replaced. 

- Is there any trick I can use to avoid losing too many macroparticles with the moving window? Even I start with a huge number of macro number of particles, the ones closer on the LHS would be lost very quickly due to the exponential profile.

Thanks again for your help,

William!

**Comment by Status-Mirror** [comment]
Hey @wnguyen1312,

This shouldn't be a problem - by default, EPOCH is set up with variable particle weighting. If you have 2 particles per cell, then each cell will be assigned two macro-particles. The weights of these macro-particles will vary to produce the density distribution you initialise.

Cheers,
Stuart
